### ISSUE TITLE
BootstrapAjaxPagingNavigator

### ISSUE THREAD

**Issue by alemi77** [l0rdn1kk0n/wicket-bootstrap#533]
Io sto usando wicket 7-M6 e quando uso BootstrapAjaxPagingNavigator viene chiamatodue volte l'evento onClick  che porta al settaggio della pagina per due volte. Nella seconda, la pagina è errata e riporta quest'errore:
org.apache.wicket.core.request.handler.ListenerInvocationNotAllowedException: Component rejected interface invocationComponent: [AjaxPagingNavigationLink [Component id = pageLink]] Listener: [RequestListenerInterface name=ILinkListener, method=public abstract void org.apache.wicket.markup.html.link.ILinkListener.onLinkClicked()]
	at org.apache.wicket.RequestListenerInterface.invoke(RequestListenerInterface.java:212)
	at org.apache.wicket.core.request.handler.ListenerInterfaceRequestHandler.invokeListener(ListenerInterfaceRequestHandler.java:241)
	at org.apache.wicket.core.request.handler.ListenerInterfaceRequestHandler.respond(ListenerInterfaceRequestHandler.java:234)
	at org.apache.wicket.request.cycle.RequestCycle$HandlerExecutor.respond(RequestCycle.java:865)
	at org.apache.wicket.request.RequestHandlerStack.execute(RequestHandlerStack.java:64)
	at org.apache.wicket.request.cycle.RequestCycle.execute(RequestCycle.java:265)
	at org.apache.wicket.request.cycle.RequestCycle.processRequest(RequestCycle.java:222)
	at org.apache.wicket.request.cycle.RequestCycle.processRequestAndDetach(RequestCycle.java:293)
	at org.apache.wicket.protocol.http.WicketFilter.processRequestCycle(WicketFilter.java:261)
	at org.apache.wicket.protocol.http.WicketFilter.processRequest(WicketFilter.java:203)
	at org.apache.wicket.protocol.http.WicketFilter.doFilter(WicketFilter.java:284)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:220)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:122)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:504)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:170)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:103)
	at org.apache.catalina.valves.AccessLogValve.invoke(AccessLogValve.java:950)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:116)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:421)
	at org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:1074)
	at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:611)
	at org.apache.tomcat.util.net.AprEndpoint$SocketProcessor.doRun(AprEndpoint.java:2466)
	at org.apache.tomcat.util.net.AprEndpoint$SocketProcessor.run(AprEndpoint.java:2455)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.lang.Thread.run(Thread.java:745)

**Comment by reiern70** [comment]
using English would be a lot more polite for no Italian speaking guys following this project :-)

**Comment by alemi77** [comment]
I'm sorry ... but I copied the text in Italian :-))) ....I am using 7-wicket M6 and when I use BootstrapAjaxPagingNavigator is called twice the onClick event. So it seems that the page is set twice. This is the error: see above

**Comment by l0rdn1kk0n** [comment]
how do you use it? please add a code example that shows this problem.

**Comment by alemi77** [comment]
package it.unimi.start.pages.comunicazioni.consultazioni;


import de.agilecoders.wicket.core.markup.html.bootstrap.button.ButtonBehavior;
import de.agilecoders.wicket.core.markup.html.bootstrap.button.Buttons;
import de.agilecoders.wicket.core.markup.html.bootstrap.button.LoadingBehavior;
import de.agilecoders.wicket.core.markup.html.bootstrap.navigation.BootstrapPagingNavigator;
import de.agilecoders.wicket.core.markup.html.bootstrap.navigation.ajax.BootstrapAjaxPagingNavigator;
import de.agilecoders.wicket.extensions.markup.html.bootstrap.form.DateTextField;
import de.agilecoders.wicket.extensions.markup.html.bootstrap.form.DateTextFieldConfig;
import it.unimi.start.MySession;
import it.unimi.start.base.BasePage;
import it.unimi.start.common.jasper.ByteArrayResourceStream;
import it.unimi.start.persistence.dao.*;
import it.unimi.start.persistence.model.ListaReportUte;
import it.unimi.start.persistence.model.VReport;
import it.unimi.start.persistence.model.aut.VamUniGcbPref;
import net.sf.jasperreports.engine.JRException;
import net.sf.jasperreports.engine.JRParameter;
import net.sf.jasperreports.engine.JasperRunManager;
import org.apache.wicket.Component;
import org.apache.wicket.MarkupContainer;
import org.apache.wicket.WicketRuntimeException;
import org.apache.wicket.ajax.AjaxRequestTarget;
import org.apache.wicket.ajax.IAjaxIndicatorAware;
import org.apache.wicket.ajax.markup.html.AjaxLink;
import org.apache.wicket.ajax.markup.html.form.AjaxButton;
import org.apache.wicket.ajax.markup.html.navigation.paging.AjaxPagingNavigator;
import org.apache.wicket.markup.html.WebMarkupContainer;
import org.apache.wicket.markup.html.basic.Label;
import org.apache.wicket.markup.html.form.Form;
import org.apache.wicket.markup.html.form.TextField;
import org.apache.wicket.markup.html.link.Link;
import org.apache.wicket.markup.html.list.ListItem;
import org.apache.wicket.markup.html.list.PageableListView;
import org.apache.wicket.model.*;
import org.apache.wicket.protocol.http.WebApplication;
import org.apache.wicket.request.handler.resource.ResourceStreamRequestHandler;
import org.apache.wicket.util.resource.IResourceStream;
import org.apache.wicket.util.time.Duration;
import org.hibernate.Session;
import org.hibernate.jdbc.Work;

import javax.servlet.ServletContext;
import java.sql.Connection;
import java.sql.SQLException;
import java.util.Date;
import java.util.List;
import java.util.Locale;
import java.util.Map;

/**
 * User: andrea Date: 06/02/14 Time: 16.55
 */
public class ConsultazioniAteneoPage extends BasePage implements IAjaxIndicatorAware {


    private AjaxButton findBtnObj;
//    private List<ListaReportUte> listRequests;
    private PageableListView requestsTable;
    private static final int numeroRighe = 5;
    private  MarkupContainer listContainer;

    private String pathToPrint;
    private String fileNameReport;
    private byte[] fileP;
    private DateTextField dtInObj;
    private DateTextField dtFiObj;

    private TextField<Integer> numElezObj;
    private Label noFoundObj;
//    private BootstrapPagingNavigator pagobj;
    private BootstrapAjaxPagingNavigator pagobj;

    private IModel<List<ListaReportUte>> listRequests;
    private AjaxLink<Void> findBtn2Obj;


    public ConsultazioniAteneoPage() {
        super();
    }

    @Override
	protected void onInitialize() {
        super.onInitialize();

        Boolean aggiornatoPref = ((VamUniGcbPrefDao) DaoFactory.get().getDao(VamUniGcbPref.class)).setPrefMatr((MySession.get().getUserMatr()), this.getClass().getSimpleName());
        final Form form = new Form("form");
        add(form);

        numElezObj = new TextField<Integer>("numElezHtml", new Model<Integer>());
        numElezObj.setOutputMarkupId(true);
        form.add(numElezObj);
        String pattern = "dd/MM/yyyy";

        dtInObj = new DateTextField("dtInHtml", Model.of(new Date(System.currentTimeMillis())),
                new DateTextFieldConfig()
                        .autoClose(true)
                        .withLanguage("it")
                        .clearButton(true)
                        .withFormat("dd/mm/yyyy")
                        .showTodayButton(DateTextFieldConfig.TodayButton.TRUE)) {
            @Override
            public IModel<String> getLabel() {
                return new StringResourceModel("dtInsHtmlErr", ConsultazioniAteneoPage.this, null);
            }
        };
        dtInObj.setOutputMarkupId(true);
        form.add(dtInObj);


        dtFiObj = new DateTextField("dtFiHtml", Model.of(new Date(System.currentTimeMillis())),
                new DateTextFieldConfig()
                        .autoClose(true)
                        .withLanguage("it")
                        .clearButton(true)
                        .withFormat("dd/mm/yyyy")
                        .showTodayButton(DateTextFieldConfig.TodayButton.TRUE)) {
            @Override
            public IModel<String> getLabel() {
                return new StringResourceModel("dtFiHtmlErr", ConsultazioniAteneoPage.this, null);
            }
        };
        dtFiObj.setOutputMarkupId(true);
        form.add(dtFiObj);

        noFoundObj = new Label("noFoundHtml", new ResourceModel("noFoundLabel"));
        noFoundObj.setOutputMarkupId(true);
        noFoundObj.setOutputMarkupPlaceholderTag(true);
        noFoundObj.setVisible(false);
        form.add(noFoundObj);


        listContainer = new WebMarkupContainer("listContainer");
        listContainer.setOutputMarkupId(true);
        listContainer.setOutputMarkupPlaceholderTag(true);


        listRequests = new LoadableDetachableModel<List<ListaReportUte>>() {

            @Override
            protected List<ListaReportUte> load() {
                return ((ListaReportUteDao) DaoFactory.get().getDao(ListaReportUte.class)).getListaReportElezUte("ConsAte", (MySession.get().getUserMatr()),
                        "StpRisultatoCons",
                        numElezObj.getModelObject() == null ? null : String.valueOf(numElezObj.getModelObject()),
                        dtInObj.getModelObject() == null ? null : dtInObj.getModelObject(),
                        dtFiObj.getModelObject() == null ? null : dtFiObj.getModelObject());
            };
        };



        findBtnObj = new AjaxButton("findBtnHtml") {
            @Override
            protected void onSubmit(AjaxRequestTarget target, Form<?> form) {
                listRequests.detach();
                noFoundObj.setVisible((listRequests.getObject().size() == 0) ? true : false);
                listContainer.setVisible((listRequests.getObject().size() == 0) ? false : true);
                target.add(noFoundObj);
                target.add(listContainer);
            }
        };
        findBtnObj.setOutputMarkupId(true);

        form.add(findBtnObj);

        requestsTable = new PageableListView<ListaReportUte>("requestsTable", listRequests, numeroRighe) {
            @Override
            protected void populateItem(final ListItem<ListaReportUte> item) {
                item.add(new Label("data", new PropertyModel(item.getModel(), "data")));
                item.add(new Label("dsId", new PropertyModel(item.getModel(), "dsId")));

                final Link pdfLink = new Link("pdf.link") {
                    @Override
                    public void onClick() {
                        try {
                            fileNameReport = item.getModelObject().getNomeReport();
                            final Map<String, Object> parameters;
                            parameters = ((VReportDao) DaoFactory.get().getDao(VReport.class)).getParametri(item.getModelObject().getId().getId(), MySession.get().getUserMatr());
                            fileP = null;
                            final ServletContext context = ((WebApplication) getApplication()).getServletContext();
                            pathToPrint = context.getRealPath("/report") + "/";

                            Locale loc_it = new Locale("it", "IT");
                            parameters.put(JRParameter.REPORT_LOCALE, loc_it);
                            parameters.put("SUBREPORT_DIR", pathToPrint);
                            parameters.put("PAR_DIR", pathToPrint);

                            Session session = Context.get().getHibernateSession();
                            session.doWork(new Work() {
                                public void execute(Connection connection) throws SQLException {
                                    try {
                                        fileP = JasperRunManager.runReportToPdf(pathToPrint + fileNameReport, parameters, connection);
                                        IResourceStream resourceStream = new ByteArrayResourceStream(fileP, "application/pdf");
                                        getRequestCycle().scheduleRequestHandlerAfterCurrent(new ResourceStreamRequestHandler(resourceStream, item.getModelObject().getDsReport().concat(".pdf")));
                                    } catch (JRException e) {
                                        new WicketRuntimeException(e);
                                    }
                                }
                            });
                            session.close();

                        } catch (Exception e) {
                            throw new WicketRuntimeException(e);
                        }
                    }
                };
                item.add(pdfLink);
            }
        };

        requestsTable.setOutputMarkupPlaceholderTag(true);
        requestsTable.setOutputMarkupId(true);
        listContainer.add(requestsTable);

        pagobj = new BootstrapAjaxPagingNavigator("pager", requestsTable) ;
        //pagobj = new BootstrapPagingNavigator("pager", requestsTable);
        listContainer.add(pagobj);
        listContainer.setVisible(false);
        form.add(listContainer);
    }
    
    public String getAjaxIndicatorMarkupId() {
        return  "loadingIndicator_id";
    }
}

**Comment by Adar** [comment]
Hi,

exactly the same problem here as I changed to currently released Wicket7, too. Even when I empty populateItems in the listview and add everything to the basic page this error occurs. But to be fair, changing it to PagingNavigator doesnt help at all (stays at page 1) so I assume there's a bug in the Wicket PagingNavigator.

**Comment by martin-g** [comment]
Please provide a minimal app demo that reproduces the problem.
Is it a problem in Wicket-Bootstrap or in Wicket?

**Comment by Adar** [comment]
I made a quick project  which demonstrates the issue https://github.com/Adar/paging-navigator-533
I tried to make it as simple as possible. This issue does not always appear on first click but it reproducibly does.

**Comment by martin-g** [comment]
I'm clicking for 2 minutes now and I cannot reproduce the issue.
Should I follow some specific steps?

**Comment by martin-g** [comment]
OK. I've reproduced it!
But it happens also with Wicket 6.20.0 + Wicket Bootstrap 0.9.12!

**Comment by Adar** [comment]
I'm guessing this is an issue in Wicket itself then?
Do you have any idea what might cause this or how to avoid it?

**Comment by martin-g** [comment]
I'm debugging it!
Stay tuned!

**Comment by martin-g** [comment]
It was a bug in Wicket Bootstrap after all.
Enjoy!

**Comment by Adar** [comment]
Great you could fix this, thanks. :+1: looking forward to the next release.
